docs/architecture/blazor-for-web-forms-developers/hosting-models.md

```diff
@@ -1,9 +1,80 @@
+
+The downloaded assemblies are normal .NET assemblies, like you would use in any other .NET app. The runtime supports .NET Standard, so you can use existing .NET Standard libraries with your Blazor WebAssembly app. However, these assemblies will still execute inside of the browser security sandbox, so some functionality may throw `PlatformNotSupportedExceptions`, like trying to access the filesystem or opening arbitrary network connections. 
+
+When the app loads, the .NET runtime is started and pointed at the app assembly. The app startup logic runs and the root components are rendered. Blazor calculates the UI updates based on the rendered output from the components and updates the DOM accordingly.
+
+![Blazor WebAssembly](media/hosting-models/blazor-webassembly.png)
+
+Blazor WebAssembly apps run purely client-side and can be deployed as static files to static site hosting solutions like GitHub Pages or Azure Static Website Hosting. .NET is not required on the server at all. To enable deep linking to different parts of the app you typically do need a routing solution on the server to redirect requests to the root of the app. For example, this can be handled using URL rewrite rules in IIS.
+
+To get all of the benefits of Blazor and full-stack .NET web development, you'll want to host your Blazor WebAssembly app with ASP.NET Core. By using .NET on both the client and server, you can easily share code and build your app using one consistent set of languages, frameworks, and tools. Blazor provides convenient templates for setting up a solution that contains both a Blazor WebAssembly app and an ASP.NET Core host project. When the solution is built, the built static files from the Blazor app are hosted by the ASP.NET Core app with fallback routing already setup.
+
+## Blazor Server apps
+
+Remember from the [Blazor architecture](architecture-comparison#Blazor) discussion that Blazor components render their output to an intermediate abstraction called a `RenderTree`. The Blazor framework then handles comparing what was rendered with what was rendered previously and updates the DOM with only what was changed. Because Blazor components are decoupled from how their rendered output gets applied, the components themselves don't have to run in the same process as the process updating the UI. In fact, they don't even have to run on the same machine.
+
+In a Blazor Server apps, the components run on the server instead of client-side in the browser. Any UI events that occur in the browser are sent to the server over a real-time connection. The events get dispatched to the correct component instances, the components render, and then the calculated UI diff is serialized back down to the browser where it is applied to the DOM.
```

Comment:
```suggestion
In Blazor Server apps, the components run on the server instead of client-side in the browser. Any UI events that occur in the browser are sent to the server over a real-time connection. The events get dispatched to the correct component instances, the components render, and then the calculated UI diff is serialized back down to the browser where it is applied to the DOM.
```